### PR TITLE
New: Allow snapshot test

### PR DESCRIPTION
**Pull request by fa93hws**
## Summary

Allow the snapshot test

## Related Issues

https://github.com/eslint/eslint/issues/11988

### CODE REVIEW

File: designs/2019-snapshot-test/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+- Start Date: 2019-07-25
+- RFC PR: Allow snapshot test
```
Comment (mysticatea) [inline designs/2019-snapshot-test/README.md]:
```suggestion
- RFC PR: https://github.com/eslint/rfcs/pull/36
```

**Comment by ljharb** [comment]
Snapshot tests are inherently brittle, and when they fail, what inevitably happens is that people rubber-stamp the snapshot updates without understanding what the real problem is. I'd recommend not including any form of snapshot testing in eslint.

**Comment by fa93hws** [comment]
> Snapshot tests are inherently brittle, and when they fail, what inevitably happens is that people rubber-stamp the snapshot updates without understanding what the real problem is. I'd recommend not including any form of snapshot testing in eslint.

I think it should be alright, because the snapshot will be checked in as well.
As a reviewer, the difference in snapshot which is not appropriate is very clear.

**Comment by ljharb** [comment]
In my experience they are very unclear, and developers (even senior ones) end up just approving snapshot changes without thorough review.

**Comment by fa93hws** [comment]
> In my experience they are very unclear, and developers (even senior ones) end up just approving snapshot changes without thorough review.

Unlike html (I feel snapshot test on html is useful as well), linting rule is simple enough to figure out what's going wrong in the snapshot test.
Especially when compared to the existing Api, snapshot test should be at least more reviewer-friendly than `{ line: 2, column: 4, columnEnd: 14 }`

**Comment by ilyavolodin** [comment]
Sorry, I'm not sold on this. Maybe I'm a minority here, but I like writing code, not some visual representation of an error. I don't find it hard to count the number of characters where the error will occur in the test. I know it's a personal preference, and others might not agree with me. However, I also do not see any barrier of creating an external package that would provide format for snapshot testing. So from that perspective, right now I'm not seeing much benefit of adding yet another format to the core.

**Comment by platinumazure** [comment]
I agree, I think this should be in a third-party package. If this is proven out and users would like to see it integrated into RuleTester, we could revisit the discussion.

**Comment by fa93hws** [comment]
A third party library SGTM

**Comment by axetroy** [comment]
Five years on, is it possible that the snapshot feature would consider restarting?

Because nodejs natively supports [snapshot testing](https://nodejs.org/docs/latest/api/test.html#snapshot-testing)

I have implemented a simple example based on this, it is very useful, hope it can be reconsidered

```
exports[`no-mix-commonjs-esm 1`] = `
## invalid(1): import bar from './bar'\\nconst foo = require('./foo')

  > Input
  
        1 | import bar from './bar'
        2 | const foo = require('./foo')

  > Error 1/1: Don't mix CommonJS (\`require\`) and ES Modules (\`import\`/\`export\`) in the same file
  
        1 | import bar from './bar'
      > 2 | const foo = require('./foo')
          |             ^^^^^^^^^^^^^^^^

## invalid(2): import * as bar from './bar'\\nconst foo = require('./foo')

  > Input
  
        1 | import * as bar from './bar'
        2 | const foo = require('./foo')

  > Error 1/1: Don't mix CommonJS (\`require\`) and ES Modules (\`import\`/\`export\`) in the same file
  
        1 | import * as bar from './bar'
      > 2 | const foo = require('./foo')
          |             ^^^^^^^^^^^^^^^^

## invalid(3): const foo = require('./foo')\\nexport default foo;

  > Input
  
        1 | const foo = require('./foo')
        2 |
        3 | export default foo;

  > Error 1/1: Don't mix CommonJS (\`require\`) and ES Modules (\`import\`/\`export\`) in the same file
  
      > 1 | const foo = require('./foo')
          |             ^^^^^^^^^^^^^^^^
        2 |
        3 | export default foo;

## invalid(4): const foo = require('./foo')\\nexport { foo };

  > Input
  
        1 | const foo = require('./foo')
        2 |
        3 | export { foo };

  > Error 1/1: Don't mix CommonJS (\`require\`) and ES Modules (\`import\`/\`export\`) in the same file
  
      > 1 | const foo = require('./foo')
          |             ^^^^^^^^^^^^^^^^
        2 |
        3 | export { foo };

## invalid(5): export const foo = require('./foo')

  > Input
  
        1 | export const foo = require('./foo')

  > Error 1/1: Don't mix CommonJS (\`require\`) and ES Modules (\`import\`/\`export\`) in the same file
  
      > 1 | export const foo = require('./foo')
          |                    ^^^^^^^^^^^^^^^^
`;

```

**Pull request closed without merging**